feat(spinner): improve build progress display using indicatif crate

navahas opened the pull request:
Closes #17

## Feature or Problem
This PR adds a build progress spinner to `wash dev` using the `indicatif` crate.  
The spinner provides visual feedback during component builds, improving the developer experience by making it clear when a build is in progress.  

## Related Issues
- #17 — Add spinner to indicate build progress in `wash dev`

## Release Information
Targeting the next release.

## Consumer Impact
- `wash dev` will now display a build spinner during component builds.
- No functional changes to build or deploy logic.

## Testing

### Acceptance or Integration
- No changes required to existing acceptance or integration tests.

### Manual Verification
- Ran `wash dev` on a sample Rust project:
  - Spinner appears during the initial build.
  - Spinner clears upon completion and shows the final build message.

navahas committed:
feat(spinner): improve build progress display using indicatif crate

Closes #17

Signed-off-by: Carlos Navajas <c.navajas4@gmail.com>

brooksmtownsend commented on Cargo.toml:
```diff
@@ -35,6 +35,7 @@ docker_credential = { version = "1.3.1", default-features = false }
 etcetera = { version = "0.10.0", default-features = false }
 figment = { version = "0.10.19", default-features = false, features = ["json", "env"] }
 hyper = { version = "1.6.0", default-features = false }
+indicatif = "0.18.0"
```
```suggestion
indicatif = { version =" 0.18.0", default-features = false }
```

Just a suggestion to keep parity, if you need any features feel free to enable them

brooksmtownsend commented on crates/wash/src/cli/dev.rs:
```diff
@@ -125,9 +127,15 @@ impl CliCommand for DevCommand {
+        spinner.set_style(
+            ProgressStyle::default_spinner()
+                .template("{spinner:.green} {msg}")
+                .unwrap(),
```
```suggestion
                .context("failed to create spinner")?,
```

Would prefer to safely bail in the case that this fails for some reason

brooksmtownsend commented on crates/wash/src/cli/dev.rs:
```diff
@@ -322,29 +333,44 @@ impl CliCommand for DevCommand {
+        dev_spinner.set_style(
+            ProgressStyle::default_spinner()
+                .template("{spinner:.green} {msg}")
+                .unwrap(),
```
```suggestion
                .context("failed to create spinner")?,
```

Would prefer to safely bail in the case that this fails for some reason

brooksmtownsend reviewed (changes requested):
Just a request to use context over unwrap, and @navahas do you have a video of how this looks when you run it? I wanted to make sure that it looks nice, I can pull it down too to test it out

navahas commented:
https://github.com/user-attachments/assets/b4cf8a2d-298a-4af1-be43-f92c51a15932

navahas commented:
I wasn’t too familiar with indicatif before, so after several attempts: printing to stderr, stdout, and trying its alternate methods like set_draw_target.

The current approach felt like the least annoying starting point.

navahas committed:
chore(spinner): address review feedback
- Use \`indicatif = { version = "0.18.0", default-features = false }\`
- Replace \`.unwrap()\` with \`.context("failed to create spinner")?\` in style setup
- Keep spinner behavior unchanged; improve error paths

Signed-off-by: Carlos Navajas <c.navajas4@gmail.com>
